File: weekly/344_/q1.py
class Solution:
    def distinctDifferenceArray(self, nums: List[int]) -> List[int]:
        ans = []
        n = len(nums)
        for i in range(n):
            s1, s2 = set(nums[:i+1]), set(nums[i+1:])
            ans.append(len(s1) - len(s2))
        return ans

File: weekly/344_/go-solutions/q1.go
func distinctDifferenceArray(nums []int) (ans []int) {
    cnt := func (arr []int) int {
        m := map[int]int{}
        for _, v := range arr {
            m[v]++
        }
        return len(m)
    }

    for i := range nums {
        ans = append(ans, cnt(nums[:i+1]) - cnt(nums[i+1:]))
    }
    return
}

